nautobot/utilities/templatetags/buttons.py

```diff
@@ -30,59 +30,57 @@ def clone_button(instance):
 
 
 @register.inclusion_tag("buttons/edit.html")
-def edit_button(instance, use_pk=False, key="slug"):
+def edit_button(instance):
     """
     Render a button to edit a model instance.
 
     Args:
         instance: Model record.
-        use_pk: If True, use the primary key instead of any specified "key" field. (Deprecated, use `key="pk"` instead)
-        key: The attribute on the model to use for reverse URL lookup.
     """
     viewname = get_route_for_model(instance, "edit")
 
-    # Assign kwargs
-    if hasattr(instance, key) and not use_pk:
-        kwargs = {key: getattr(instance, key)}
-    else:
-        kwargs = {"pk": instance.pk}
+    # We try different lookups to get a valid reverse url
+    lookups = ["pk", "slug"]
```

Comment:
I might be missing a lookup key here. But I do not see any urls that are using `name` or `key` for lookup in develop

Reply:
Maybe include `key` just because we know it'll be needed in `next` for the time being?